crypto/x509/x509_vfy.c

```diff
@@ -3094,8 +3091,11 @@ static int build_chain(X509_STORE_CTX *ctx)
             }
             curr = sk_X509_value(ctx->chain, i - 1);
 
-            ok = num > depth ? 0 : get_issuer(&issuer, ctx, curr);
+            /* Stop immediately with success if the current cert is trusted */
+            if ((trust = check_trust(ctx, num)) == X509_TRUST_TRUSTED)
+                break;
```

Comment:
Didn't we finally agree that this is not quite right? Was that in a separate PR that overlaps with this one?

Reply:
Right. It was in this PR, namely in https://github.com/openssl/openssl/pull/13748#discussion_r581016359 - please place any further comments on that specific topic there.
I just haven't found the time yet to adapt this PR accordingly.

What you could do in the meantime is to answer to my latest reactions to your comments in #13735. This has been waiting for your input since end of January,
and would be great if we can finalize that soon.

Reply:
**Update:** @vdukhovni, I've just done that as discussed above.